scripts/challenge_week.py

```diff
@@ -0,0 +1,260 @@
+        self.df_submit = pd.read_csv(f'{self.path}/data/task{set}/teamname_set{set}.csv')
+
+        return self.df_demand, self.df_pv, self.df_weather, self.df_submit
+
+    def merge_dataframes(self):
+        '''Creates two combined dataframes from raw task week data.
+
+        Returns:
+            df_train: dataframe containing all weather data, pv_power data, and demand data
+                      given for the task week. There will be missing weather data at this point,
+                      becuase the weather data is hourly.
+            df_taskweek: dataframe containing all the wather data that can be used for
+                         prediction of the wekk of interest.
+
+        '''
+        # Check 'datetime' columns are equal in demand and pv dfs, this must be true
+        assert self.df_demand['datetime'].equals(self.df_pv['datetime']), "Demand and PV train sets have different timestamps"
+
+        # Merge the demand and pv_power dataset together into one dataframe
+        self.df_train = self.df_pv.merge(right=self.df_demand,
+                                         on='datetime',
+                                         how='left')
+
+        # Create a weather sets that matches the date range of pv_demand set and submit set
+        # Set DatetimeIndex
+        self.df_weather.set_index('datetime',
+                                  inplace=True)
+
+        # Create weather dataframe with matching timestamps with training/testing data
+        start_date = self.df_train.loc[0, 'datetime']
+        end_date = self.df_train.loc[len(self.df_train)-1, 'datetime']
+        df_weather_train = self.df_weather.loc[start_date:end_date]
+
+        # Create weather dataframe with matching timestamps with task week submission data
+        start_date = self.df_submit.loc[0, 'datetime']
+        end_date = self.df_submit.loc[len(self.df_submit)-1, 'datetime']
+        df_weather_taskweek = self.df_weather.loc[start_date:end_date]
+
+        # Join pv_demand set with weather_set0 set to complete the model dataframe
+        self.df_train = self.df_train.merge(right=df_weather_train,
+                                            on='datetime',
+                                            how='outer')
+        self.df_taskweek = self.df_submit.merge(right=df_weather_taskweek,
+                                                  on='datetime',
+                                                  how='outer')
+
+        return self.df_train, self.df_taskweek
+
+    def impute_weather(self, lst_cols, df):
+        '''Imputes the missing data for the weather paramters using the mean of the
+        timesteps before and after.
+
+        Args:
+            lst_cols(list): list of weather columns to impute
+            df: dataframe to which to pply the transformation
+        Returns:
+            df: transformed df
+        '''
+        for c in range(0, len(lst_cols)):
```

Comment:
Don't have to use it, but pandas does have some neat functions to facilitate this, `bfill` and `ffill`, and can be used in combination to get an average as per https://stackoverflow.com/questions/44102794/input-missed-values-with-mean-of-nearest-neighbors-in-column

Reply:
I was wondering, surely someone has a pandas method written up to do this better than what I have....i shall check this out